File: solutions/python/guidos-gorgeous-lasagna/1/lasagna.py
"""Functions used in preparing Guido's gorgeous lasagna.

Learn about Guido, the creator of the Python language: https://en.wikipedia.org/wiki/Guido_van_Rossum
"""

EXPECTED_BAKE_TIME = 40

PREPARATION_TIME = 2

def bake_time_remaining(elapsed_bake_time):
    """Calculate the bake time remaining.

    :param elapsed_bake_time: int - baking time already elapsed.
    :return: int - remaining bake time (in minutes) derived from 'EXPECTED_BAKE_TIME'.

    Function that takes the actual minutes the lasagna has been in the oven as
    an argument and returns how many minutes the lasagna still needs to bake
    based on the `EXPECTED_BAKE_TIME`.
    """
    result = EXPECTED_BAKE_TIME - elapsed_bake_time
    if result < 0:
        return 0
    return result


def preparation_time_in_minutes(layers):
    return layers * PREPARATION_TIME


def elapsed_time_in_minutes(number_of_layers, elapsed_bake_time):
    return preparation_time_in_minutes(number_of_layers) + bake_time_remaining(elapsed_bake_time)

File: solutions/go/lasagna/1/lasagna.go
package lasagna

// TODO: define the 'OvenTime' constant

var OvenTime = 40;

// TODO: define the 'RemainingOvenTime()' function

func RemainingOvenTime(minutes int) int {

    return OvenTime - minutes;
    
}
// TODO: define the 'PreparationTime()' function

func PreparationTime(layers int) int {
    return layers * 2;
    
}
// TODO: define the 'ElapsedTime()' function

func ElapsedTime(layers int, minutes int) int {

    return minutes + PreparationTime(layers);

}